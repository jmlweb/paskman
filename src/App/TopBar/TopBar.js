import React from 'react';
import PT from 'prop-types';
import styled from 'styled-components';
import colors from '../../styles/colors';
import spacing from '../../styles/spacing';
import topBarHeight from '../../styles/topBarHeight';
import Logo from '../../components/Logo/Logo';
import TopBarBtn from './TopBarBtn/TopBarBtn';
import TopBarMenu from './TopBarMenu/TopBarMenu';
import { RoutesPT } from '../../propTypes';

const StyledTopBar = styled.div`
  background: ${colors.primary};
  color: ${colors.secondary};
`;

const StyledWrapper = styled.div`
  align-items: center;
  display: flex;
  height: ${topBarHeight};
  justify-content: space-between;
  padding: 0 ${spacing.sm};
`;

const StyledNav = styled.div``;

const TopBar = ({ menuOpen, routes, topBarToggleMenu }) => (
  <StyledTopBar>
    <StyledWrapper>
      <Logo />
      <StyledNav>
        <TopBarBtn menuOpen={menuOpen} toggleMenu={topBarToggleMenu} />
        <TopBarMenu menuOpen={menuOpen} toggleMenu={topBarToggleMenu} routes={routes} />
      </StyledNav>
    </StyledWrapper>
  </StyledTopBar>
);

TopBar.defaultProps = {
  menuOpen: false,
  routes: [],
};

TopBar.propTypes = {
  menuOpen: PT.bool,
  routes: RoutesPT,
  topBarToggleMenu: PT.func.isRequired,
};

export default TopBar;
