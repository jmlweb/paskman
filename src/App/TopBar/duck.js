import {
  createAction,
  handleAction,
} from 'redux-actions';
import stateMock from '../../stateMock';

/**
 * CONSTANTS
 */
export const TOP_BAR_TOGGLE_MENU = 'TOP_BAR/TOGGLE_MENU';

/**
 * ACTIONS
 */

export const topBarToggleMenu = createAction(TOP_BAR_TOGGLE_MENU);

/**
 * REDUCER
 */
export const initialState = stateMock.app.topBar;

const reducer = handleAction(
  TOP_BAR_TOGGLE_MENU,
  state => ({ ...state, menuOpen: !state.menuOpen }),
  initialState,
);

export default reducer;
