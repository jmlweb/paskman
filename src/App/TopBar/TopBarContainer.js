// @flow
import { connect } from 'react-redux';
import { topBarToggleMenu } from './duck';
import TopBar from './TopBar';

export function mapStateToProps(state: any) {
  return {
    menuOpen: state.app.topBar.menuOpen,
  }
};

const TopBarContainer = connect(mapStateToProps, {
  topBarToggleMenu,
})(TopBar);

export default TopBarContainer;
