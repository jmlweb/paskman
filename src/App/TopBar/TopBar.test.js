import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import TopBar from './TopBar';

describe('TopBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<TopBar menuOpen={true} topBarToggleMenu={jest.fn()} routes={[]} />);
  });
  it('Render the component', () => {
    expect(wrapper.find(TopBar).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
