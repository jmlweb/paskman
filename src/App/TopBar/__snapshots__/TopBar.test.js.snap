// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Matchs snapshot 1`] = `
<div
  className="sc-EHOje gITUcI"
>
  <div
    className="sc-bZQynM IknJE"
  >
    <span
      className="sc-bdVaJa kyIXQx"
    >
      <svg
        height="29"
        viewBox="0 0 40 29"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#223177"
          fillRule="evenodd"
          transform="translate(0 .8)"
        >
          <circle
            cx="14"
            cy="14"
            opacity=".626"
            r="14"
          />
          <circle
            cx="28.7"
            cy="13.3"
            opacity=".725"
            r="10.5"
          />
        </g>
      </svg>
      <span
        className="sc-bwzfXH FwGDt"
      >
        PASKMAN
      </span>
    </span>
    <div
      className="sc-gzVnrw ibZhkv"
    >
      <button
        aria-label="Close menu"
        className="sc-htpNat hMfwUI"
        onClick={[Function]}
      >
        <svg
          height="35"
          viewBox="0 0 35 35"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#1E2F76"
            fillRule="nonzero"
          >
            <path
              d="M17.5 35C7.85 35 0 27.15 0 17.5S7.85 0 17.5 0 35 7.85 35 17.5 27.15 35 17.5 35zm0-32.813c-8.444 0-15.313 6.87-15.313 15.313 0 8.444 6.87 15.313 15.313 15.313 8.444 0 15.313-6.87 15.313-15.313 0-8.444-6.87-15.313-15.313-15.313z"
            />
            <path
              d="M22.97 24.063c-.28 0-.56-.108-.775-.32L11.258 12.804c-.428-.428-.428-1.12 0-1.547.428-.428 1.12-.428 1.547 0l10.937 10.937c.428.428.428 1.12 0 1.547-.213.213-.493.32-.773.32z"
            />
            <path
              d="M22.97 24.063c-.28 0-.56-.108-.775-.32L11.258 12.804c-.428-.428-.428-1.12 0-1.547.428-.428 1.12-.428 1.547 0l10.937 10.937c.428.428.428 1.12 0 1.547-.213.213-.493.32-.773.32z"
            />
            <path
              d="M12.03 24.063c-.28 0-.56-.108-.772-.32-.428-.43-.428-1.12 0-1.548l10.937-10.937c.428-.428 1.12-.428 1.547 0 .428.428.428 1.12 0 1.547L12.805 23.742c-.214.213-.494.32-.774.32z"
            />
          </g>
        </svg>
      </button>
      <nav
        className="sc-ifAKCX fKhonf"
      />
    </div>
  </div>
</div>
`;
